packages/vite/src/node/build.ts

```diff
@@ -894,7 +961,8 @@ async function buildEnvironment(
     }
     throw e
   } finally {
-    if (bundle) await bundle.close()
+    // close the bundle will make the rolldown hmr invalid, so dev build need to disable it.
+    if (bundle && !server) await bundle.close()
```

Comment:
We should call `await bundle.close()` before exiting the process like:
https://github.com/vitejs/rolldown-vite/blob/9cbf7c8021a254e9181530170d9240634b6ca3bb/packages/vite/src/node/server/index.ts#L778-L785

Reply:
FIxed